src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java

```diff
@@ -121,100 +121,32 @@ Because our prometheus queries use a range vector (look back) of 1h, we need to
-                      String billingAccountId = labels.get("billing_marketplace_account");
-                      String account = labels.get("ebs_account");
-
-                      // For the openshift metrics, we expect our results to be a 'matrix'
-                      // vector [(instant_time,value), ...] so we only look at the result's
-                      // getValues() data.
-                      for (List<BigDecimal> measurement : item.getValues()) {
-                        BigDecimal time = measurement.get(0);
-                        BigDecimal value = measurement.get(1);
-
-                        OffsetDateTime eventTermDate = clock.dateFromUnix(time);
-                        // Need to subtract the step because we are averaging and the metric value
-                        // actually represents the end of the measured period. The start of the
-                        // event should be at the beginning.
-                        OffsetDateTime eventDate =
-                            eventTermDate.minusSeconds(metricProperties.getStep());
-
-                        Event event =
-                            createOrUpdateEvent(
-                                existing,
-                                account,
-                                orgId,
-                                tagMetric.get().getId(),
-                                clusterId,
-                                sla,
-                                usage,
-                                role,
-                                eventDate,
-                                eventTermDate,
-                                subDefOptional.get().getServiceType(),
-                                billingProvider,
-                                billingAccountId,
-                                Uom.fromValue(tagMetric.get().getId()),
-                                value,
-                                tag);
-                        events.putIfAbsent(EventKey.fromEvent(event), event);
-                      }
-                    });
+                    item ->
+                        sendEventFromData(
+                            item, eventsSent, tag, orgId, tagMetric.get(), subDefOptional.get()));
 
             if (StatusType.ERROR.equals(metricData.getStatus())) {
               throw new MeteringException(
                   String.format(
                       "Unable to fetch %s %s metrics: %s", tag, metric, metricData.getError()));
             }
 
-            eventController.saveAll(events.values());
-            log.info("Persisted {} events for {} {} metrics.", events.size(), tag, metric);
+            log.info("Sent {} events for {} {} metrics.", eventsSent.size(), tag, metric);
+            // Send event to delete any stale events found during the period only if
+            // there were events sent.
+            if (!eventsSent.isEmpty()) {
+              sendCleanUpEvent(tag, orgId, tagMetric.get(), start);
```

Comment:
One important note here is the `start` timestamp to use since the metrics with an older timestamp will be considered stale and hence deleted. 
Therefore, is the current logic correct? Or should we use the shortest timestamp found from the metrics instead?